WooCommerce/Classes/POS/Utils/StockFormatter.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import struct Yosemite.POSSimpleProduct
+
+struct StockFormatter {
+    static func stockStatusLabel(for product: POSSimpleProduct) -> String {
+        switch product.manageStock {
+        case false:
+            return product.productStockStatus.description
+        case true:
+            guard let stock = product.stockQuantity else {
+                return product.productStockStatus.description
+            }
+            if stock <= 0 {
+                return Localization.outOfStock
+            } else {
+                return String.localizedStringWithFormat(Localization.inStockWithQuantity, "\(stock)")
+            }
+        }
+    }
+
+    enum Localization {
+        static let outOfStock = NSLocalizedString(
+            "pos.stockStatusLabel.outofstock",
+            value: "Out of stock",
+            comment: "Label to be displayed in the product's card when out of stock")
+
+        static let inStockWithQuantity = NSLocalizedString(
+            "pos.stockStatusLabel.outofstock",
+            value: "%@ in stock",
```

Comment:
nit: for parameters in a localizable string, I believe we use [positional placeholders](https://github.com/woocommerce/woocommerce-ios/blob/864c68462990b7627f03a3379808686fe809d126/docs/localization.md#always-use-positional-placeholders). And it can take non-string types. Plus a mention of the parameter in the string comment. Suggested diff is assuming the stock quantity is Int64, can update it to the expected type (Double?).
```suggestion
            value: "%1$ld in stock",
```

Reply:
TIL: _"Always include the positional index of parameters, even if you only have one placeholder in your string."_ updated https://github.com/woocommerce/woocommerce-ios/pull/15670/commits/2b86c074d00b13263691638a8153a1b82f755b18